presto-orc/src/main/java/io/prestosql/orc/OrcWriter.java

```diff
@@ -468,6 +469,42 @@ public void close()
         }
     }
 
+    public enum OrcOperation
+    {
+        NONE(-1),
+        INSERT(0),
+        DELETE(2);
+
+        private final int operationNumber;
+
+        OrcOperation(int operationNumber)
+        {
+            this.operationNumber = operationNumber;
+        }
+
+        public int getOperationNumber()
+        {
+            return operationNumber;
+        }
+    }
+
+    public void updateUserMetadata(long writeId, int bucket, OrcOperation operation)
```

Comment:
This code is specific to Hive ACID and belongs in the Hive connector. The ORC writer should have a simple method to add/update metadata entries in the map.

Reply:
It required a bit of remodularization, but done.